### DISCUSSION TITLE
One-time execution when connected to a database

### DISCUSSION THREAD

**Discussion by s-ohnishi** [phalcon/cphalcon#16090] [Q&A] [answered]
I am using Phalcon 5.0.0RC4.

There is no problem in the local test environment, but in the production environment only the time zone of the database is UTC and cannot be changed.
Therefore, I would like to execute `SET time_zone = '+9:00'` only once when connecting, where is the best place to do it?
Is it the place where the db service is registered in the container, or the initialize() of the model class?
Or is there another?

**Comment by niden** [comment, marked as answer]
You can do it when registering your database provider.

```php
$container->setShared(
    'db',
    function () {
        $options = [
            'host'     => envValue('DATA_API_MYSQL_HOST', 'localhost'),
            'username' => envValue('DATA_API_MYSQL_USER', 'phalcon'),
            'password' => envValue('DATA_API_MYSQL_PASS', ''),
            'dbname'   => envValue('DATA_API_MYSQL_NAME', 'phalcon_api'),
        ];

        $connection = new Mysql($options);
        // Set everything to UTF8
        $connection->execute('SET NAMES utf8mb4', []);
        // Set time to UTC
        $connection->execute("SET time_zone='+9:00'", []);

        return $connection;
    }
);
```

**Reply by s-ohnishi** [reply]
This should be done when registering for the service. Thank you.
For a long time, I specified the charset in my DSN, as I was informed that using `SET NAMES ~` could lead to vulnerabilities when using libmysqlclient.
Even in Phalcon, I used it conveniently because I can specify the DSN collectively in an array (including the character set).
However, since it has already been replaced by mysqlnd, there may be no problem using `SET NAMES ~`.
But, I feel that it is easier to set with DSN....
